ztoc/ztoc.go

```diff
@@ -35,20 +35,22 @@ import (
 // (2). zinfo (`CompressionInfo`): a collection of "checkpoints" of the
 // state of the compression engine at various points in the layer.
 type Ztoc struct {
+	TOC
+	CompressionInfo
+
 	Version                 string
 	BuildToolIdentifier     string
 	CompressedArchiveSize   compression.Offset
 	UncompressedArchiveSize compression.Offset
-	TOC                     TOC
-	CompressionInfo         CompressionInfo
 }
 
 // CompressionInfo is the "zinfo" part of ztoc including the `Checkpoints` data
 // and other metadata such as all span digests.
 type CompressionInfo struct {
-	MaxSpanID   compression.SpanID //The total number of spans in Ztoc - 1
-	SpanDigests []digest.Digest
-	Checkpoints []byte
+	MaxSpanID            compression.SpanID //The total number of spans in Ztoc - 1
+	SpanDigests          []digest.Digest
+	Checkpoints          []byte
+	CompressionAlgorithm string
```

Comment:
Why a string instead of the byte that's used to store this in the fb?

Reply:
containerd returns compression algorithms as plain string ("gzip", "zstd"), so we just use the same (string) to keep it consistent with containerd and I didn't feel extra benefit of defining a new type for that ourselves (e.g. `type CompressionAlgorithm string`).

Also fb is only used to (de)serilize `Ztoc` struct from/to flatbuf (and has its own [Compression type definition](https://github.com/awslabs/soci-snapshotter/blob/main/ztoc/fbs/ztoc/CompressionAlgorithm.go)), but Ztoc is used in many places of soci. I don't think we should use fb-specific types within Ztoc. So the conversion is made only in ztoc_marshaler.go.